tools/codesign_mac_executable.py

```diff
@@ -0,0 +1,232 @@
+#
+# Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+#
+
+# Please refer to https://medium.com/@parthnaik92/codesign-and-notarize-unreal-engine-mac-builds-for-distribution-outside-of-the-mac-store-d2f6e444f3a7
+# for pre-requisites and setup your system before trying to run this file.
+
+import argparse
+import os
+import shutil
+import subprocess
+import time
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--macos_executable", required=True)
+    parser.add_argument("--entitlements_file", required=True)
+    parser.add_argument("--developer_id", required=True)
+    parser.add_argument("--output_dir", required=True)
+    parser.add_argument("--apple_username", required=True)
+    parser.add_argument("--apple_password", required=True)
+    parser.add_argument("--wait_time_seconds", type=float, default=600.0)
+    args = parser.parse_args()
+
+    assert os.path.exists(args.macos_executable)
+    radio_effect_units = [
+        os.path.join(args.macos_executable, "Contents", "Resources", "RadioEffectUnit.component"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Build", "Mac", "RadioEffectUnit")]
+
+    print(f"[SPEAR | codesign_mac_executable.py] Removing {radio_effect_units[0]}.")
+    shutil.rmtree(radio_effect_units[0], ignore_errors=True)
+
+    print(f"[SPEAR | codesign_mac_executable.py] Removing {radio_effect_units[1]}.")
+    shutil.rmtree(radio_effect_units[1], ignore_errors=True)
+
+    print(f"[SPEAR | codesign_mac_executable.py] Changing rpaths...")
+
+    # change current working directory to add relative rpaths 
+    cwd = os.getcwd()
+    os.chdir(os.path.join(args.macos_executable, ".."))
+
+    executable_name = os.path.basename(args.macos_executable)
+
+    cmd = [
+        "install_name_tool",
+        "-add_rpath",
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "Ogg", "Mac"),
```

Comment:
Have you tested this on multiple mac machines? I think this may actually prevent the executable from running on other machines because these rpaths will only be valid for the machine where you're executing the script.

Reply:
As an aside, I tested out something similar, instead of using `executable_name`, I used `args.executable` i.e. an absolute path and placed the binary in a different location from that of `args.executable` and it did not work on the same machine itself. This is why I'm confident that the relative path works.

Reply:
Ok that makes sense. I think I misunderstood the previous code.